tasks/bootstrap_local_vm/05_add_host.yml

```diff
@@ -126,16 +126,47 @@
     retries: 120
     delay: 5
   - debug: var=host_result_up_check
-  - name: Check host status
-    fail:
-      msg: >
-        The host has been set in non_operational status,
-        please check engine logs,
-        fix accordingly and re-deploy.
+  - name: Handle non_operational failure
+    block:
+      - set_fact: host_id={{ host_result_up_check.ansible_facts.ovirt_hosts[0].id }}
+      - name: Return all events of type greater then error
+        ovirt_event_facts:
+          auth: "{{ ovirt_auth }}"
+          search: "severity>=error"
+        register: error_events
+
+      - name: Generate the error message from the engine events.
+        set_fact:
+          error_description: >-
+            {% for event in  error_events.ansible_facts.ovirt_events | groupby('code') %}
+            {% if  event[1][0].host.id  == host_id %}
+            code {{ event[0] }}: {{ event[1][0].description }},
+            {% endif %}
+            {% endfor %}
+        ignore_errors: true
+
+      - name: Fail with event extracted specific error
```

Comment:
Complicated task name. rephrase to:
Fail with error description

Try to read those names from a user perspective.
Also, don't explain in the task name what you did early or how you got the error you are printing now.
The task name suppose to describe what the task does - in this case, printing the error.

Reply:
done